aten/src/ATen/core/function_schema.h

```diff
@@ -72,6 +72,14 @@ struct Argument {
           name(),
           "' to be of type 'Tensor' ",
           "because it was not annotated with an explicit type.\n");
+    } else if(type()->repr_str() == "float" || type()->repr_str() == "int") {
+      if (actual_type == "int" || actual_type == "float") {
+      inferred_type_hint = c10::str(
+          "Did you forget to assign a '",
```

Comment:
This is a lot better than last version, thanks! I think we can be even clearer here though to let users know about their misusing float/int literals, something like "Maybe you assigned a xxx literal (1.0) rather than yyy literal (1)"?